crawl-ref/source/job-data.h

```diff
@@ -77,6 +77,17 @@ static const map<job_type, job_def> job_data =
       { SK_THROWING, 2 }, { SK_WEAPON, 2 }, },
 } },
 
+{ JOB_BATTLEMAGE, {
+    "Ba", "Battlemage",
+    4, 4, 4,
+    { SP_TENGU, SP_KOBOLD, SP_BASE_DRACONIAN, SP_NAGA, SP_TROLL,
+      SP_MERFOLK, SP_DEMONSPAWN, },
+    { "leather armour plus:1", "book of War Wizardry" },
+    WCHOICE_NONE,
+    { { SK_FIGHTING, 1}, { SK_UNARMED_COMBAT, 3 }, { SK_DODGING, 2 },
+      { SK_SPELLCASTING, 2 }, { SK_TRANSMUTATIONS, 1 }, { SK_HEXES, 1 }, },
```

Comment:
After a little more thought, and since I don't really have anywhere else to write about this stuff, here are some thoughts on a new direction for hellcrawl magic (sry this has little to do w/ your pull request):

- Remove conjurations school and most conjurations-only spells, maybe keep ood, remove conjurer background
- Crosstraining between fire/earth, ice/air, and necromancy/summoning
- Recombine charms and hexes into enchantments
- Remove spellcasting skill, adjust effect of XL to compensate (and/or use total spell school investment to determine mp bonus?)
- Fire/earth becomes more or less exclusively about direct damage
- Ice/air becomes more charms/summoning oriented, books contain fewer direct damage spells
- Vehumet-exclusive direct damage spells. Might include: bolt of cold, ozo's refrigeration, CBL, clight, tornado, glaciate, shatter, bolt of draining, airstrike, and other mid-to-high level air and ice direct damage spells.
- Ice/air elementalist background with artifact starting book something like: freeze, swiftness, rmsl, static discharge, ozo's armor, ice beast, lightning spire, throw icicle 
- Fire/earth elementalist background with artifact starting book something like: flame tongue, stone arrow, cflame, passwall, petrify, sticky flame, fireball
- Necro/summoning ally-oriented background: animate skeleton, summon small mammal, summon imp, recall, animate dead, control undead, summon lesser demon (a new spell would be cool here, e.g. "spirit animal" l3 summon/necro, summons a spectral animal)
- Enchanter/necro (not sure how good an idea this is tbh): infusion, pain, slow, vampiric draining, confuse, tukima's, agony
- Permacharms, permasummons, permadurations w/ mp reserve mechanics -- possibly poachable from circus animals to some extent
- Some guaranteed mid-level book source would be cool, either as a shop or vault. Maybe just always generate a level 5/6 art book somewhere, e.g. temple, vaults entrance vault, orc entrance etc.
- A skald/wizard background. Maybe leather, buckler, skald weapon choice with book: Infusion, rmsl, blink, song of slaying, meph or dazzling spray, spec weap, ozo's armor